addons/website/models/assets.py

```diff
@@ -59,6 +62,20 @@ def _get_custom_view(self, custom_url, op='='):
         res = super(Assets, self)._get_custom_view(custom_url, op=op)
         return res.with_context(website_id=website.id).filter_duplicate()
 
+    def get_all_custom_attachments(self, urls):
+        """
+        Fetch all the ir.attachment records related to given URLs.
+
+        Params:
+            urls (str[]): list of urls
+
+        Returns:
+            ir.attachment(): attachment records related to the given URLs.
+        """
+        if not self.env.user.has_group('website.group_website_designer'):
+            raise werkzeug.exceptions.Forbidden()
+        return super(Assets, self.sudo()).get_all_custom_attachments(urls)
+
```

Comment:
That whole block is useless?
The sudo here is not doing anything since you added it in `_get_custom_attachment` later.
Not only it is useless but it is done on a public method, so if you can get rid of it it is always better !

```suggestion
```